src/api/product/controller/product.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Param,
+  Post,
+  Patch,
+  Delete,
+  UseGuards,
+} from '@nestjs/common';
+import { ProductDto } from '../_dto/product.dto';
+import { ProductService } from '../service/product.service';
+import { ApiTags } from '@nestjs/swagger';
+import { AuthGuard } from 'src/api/auth/guards/auth.guard';
+
+@ApiTags('Product')
+@Controller('product')
+export class ProductController {
+  constructor(private readonly productService: ProductService) {}
+
+  @Post()
+  @UseGuards(AuthGuard)
+  create(@Body() productDto: ProductDto) {
+    return this.productService.createProduct(productDto);
+  }
+
+  @Get()
+  getAllProducts() {
+    return this.productService.getAllProducts();
+  }
+
+  @Patch(':id')
+  @UseGuards(AuthGuard)
+  update(@Param('id') id: string, @Body() productDto: ProductDto) {
+    return this.productService.updateProduct(productDto, id);
+  }
+
+  @Get('/list-by-category/:name')
```

Comment:
let's rename this to /category/:name

http://stayyoung.com/api/products/category/face-wash GET
it will be more clear